packages/gestalt/src/boxTransforms.js

```diff
@@ -0,0 +1,648 @@
+            : identity(),
+        ])
+      )
+    );
+  },
+
+  display,
+  column,
+  direction,
+
+  smDisplay,
+  smColumn,
+  smDirection,
+
+  mdDisplay,
+  mdColumn,
+  mdDirection,
+
+  lgDisplay,
+  lgColumn,
+  lgDirection,
+
+  alignContent,
+  alignItems,
+  alignSelf,
+  bottom,
+  borderSize,
+  color,
+  fit,
+  flex,
+  height,
+  justifyContent,
+  left,
+
+  margin,
+  marginTop,
+  marginRight,
+  marginBottom,
+  marginLeft,
+  marginStart,
+  marginEnd,
+
+  smMargin,
+  smMarginTop,
+  smMarginRight,
+  smMarginBottom,
+  smMarginLeft,
+  smMarginStart,
+  smMarginEnd,
+
+  mdMargin,
+  mdMarginTop,
+  mdMarginRight,
+  mdMarginBottom,
+  mdMarginLeft,
+  mdMarginStart,
+  mdMarginEnd,
+
+  lgMargin,
+  lgMarginTop,
+  lgMarginRight,
+  lgMarginBottom,
+  lgMarginLeft,
+  lgMarginStart,
+  lgMarginEnd,
+
+  maxHeight,
+  maxWidth,
+  minHeight,
+  minWidth,
+  opacity,
+  overflow,
+
+  padding,
+  paddingX,
+  paddingY,
+  smPadding,
+  smPaddingX,
+  smPaddingY,
+  mdPadding,
+  mdPaddingX,
+  mdPaddingY,
+  lgPadding,
+  lgPaddingX,
+  lgPaddingY,
+
+  position,
+  right,
+  rounding,
+  top,
+  userSelect,
+  width,
+  wrap,
+  dangerouslySetInlineStyle,
+  zIndex,
+};
+
+/*
+
+# Style Builder
+
+This is where it all comes together. This function takes the base styles for the component,
+the component's props, and any disallowed props. It outputs the passthrough props (after
+removing disallowed and given props), as well as an object with the combined classNames
+and styles from the given props.
+
+*/
+
+// $FlowExpectedError[unclear-type]
+export function buildStyles<T: Object>(
+  baseStyles: string,
+  props: T,
+  disallowedProps?: Array<string>
+): {|
+  passthroughProps: T,
+  propsStyles: ToPropsOutput,
+|} {
+  // Flow can't reason about the constant nature of Object.keys so we can't use
+  // a functional (reduce) style here.
+
+  // All Box's are box-sized by default, so we start off building up the styles
+  // to be applied with a Box base class.
+  let s = fromClassName(baseStyles);
+
+  // Init the list of props we'll omit from passthrough. We'll add to this
+  // list as we match props against the transforms list.
+  const omitProps = [...(disallowedProps ?? [])];
+
+  // This loops through each property and if it exists in the previously
+  // defined transform map, concatenates the resulting styles to the base
+  // styles. If there's a match, we also don't pass through that property. This
+  // means Box's runtime is only dependent on the number of properties passed
+  // to it (which is typically small) instead of the total number of possible
+  // properties (~30 or so). While it may ~feel~ like Box is inefficient, its
+  // biggest performance impact is on startup time because there's so much code
+  // here.
+
+  // eslint-disable-next-line no-restricted-syntax
+  for (const prop in props) {
+    if (
+      Object.prototype.hasOwnProperty.call(propToFn, prop) &&
+      !omitProps.includes(prop)
```

Comment:
this is new, ensuring that disallowed props aren't just not passed through, but also aren't applied (this is what keeps the disallowed Box props from working)